### DISCUSSION TITLE
Understanding chunk size for 3D datasets

### DISCUSSION THREAD

**Discussion by kushalkolar** [zarr-developers/zarr-python#1400] [General]
Trying to correctly understand optimal chunk size, from the docs:

> The optimal chunk shape will depend on how you want to access the data. E.g., for a 2-dimensional array, if you only ever take slices along the first dimension, then chunk across the second dimenson. If you know you want to chunk across an entire dimension you can use None or -1 within the chunks argument, e.g.:

https://zarr.readthedocs.io/en/stable/tutorial.html#chunk-optimizations

If my data are 3D, `[dim0, dim1, dim2]`, and I only ever slice along `dim0`, such as `a[0, :, :]` or `a[0:10, :, :]`, then would it be optimal to set `chunks=(n, None, None)` where `n` is my typical slice length?

Thanks!

**Comment by d-v-b** [comment]
yep, i think that would be reasonable! I would recommend trying this out with some sample data to see if it's ergonomic.

**Comment by kushalkolar** [comment]
It seems like `None` is no longer used to specify no chunking along a specific dimension is no longer a part of the API. How is this specified with the latest version of `zarr`? Thanks!
